src/main/java/wolox/training/models/Book.java

```diff
@@ -0,0 +1,294 @@
+        assertImage(image);
+        assertTitle(title);
+        assertSubtitle(subtitle);
+        assertPublisher(publisher);
+        assertYear(year);
+        assertPages(pages);
+        assertIsbn(isbn);
+        this.id = 0; // Will be set when saving by JPA provider
+        this.genre = genre;
+        this.author = author;
+        this.image = image;
+        this.title = title;
+        this.subtitle = subtitle;
+        this.publisher = publisher;
+        this.year = year;
+        this.pages = pages;
+        this.isbn = isbn;
+    }
+
+
+    /**
+     * @return The book's id.
+     */
+    public long getId() {
+        return id;
+    }
+
+    /**
+     * @return The book's genre.
+     */
+    public String getGenre() {
+        return genre;
+    }
+
+    /**
+     * @return The book's author.
+     */
+    public String getAuthor() {
+        return author;
+    }
+
+    /**
+     * @return The book's image.
+     */
+    public String getImage() {
+        return image;
+    }
+
+    /**
+     * @return The book's title.
+     */
+    public String getTitle() {
+        return title;
+    }
+
+    /**
+     * @return The book's subtitle.
+     */
+    public String getSubtitle() {
+        return subtitle;
+    }
+
+    /**
+     * @return The book's publisher.
+     */
+    public String getPublisher() {
+        return publisher;
+    }
+
+    /**
+     * @return The year the book was/is/will be published.
+     */
+    public String getYear() {
+        return year;
+    }
+
+    /**
+     * @return The amount of pages in the book.
+     */
+    public int getPages() {
+        return pages;
+    }
+
+    /**
+     * @return The book's ISBN.
+     */
+    public String getIsbn() {
+        return isbn;
+    }
+
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Book)) {
+            return false;
+        }
+        final Book book = (Book) o;
+        return id == book.id;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    @Override
+    public String toString() {
+        return "Book{" +
+            "id=" + id +
+            ", genre='" + genre + '\'' +
+            ", author='" + author + '\'' +
+            ", image='" + image + '\'' +
+            ", title='" + title + '\'' +
+            ", subtitle='" + subtitle + '\'' +
+            ", publisher='" + publisher + '\'' +
+            ", year='" + year + '\'' +
+            ", pages=" + pages +
+            ", isbn='" + isbn + '\'' +
+            '}';
+    }
+
+
+    /**
+     * Asserts the given {@code author}.
+     *
+     * @param author The author value to be asserted.
+     * @throws IllegalArgumentException If the given {@code author} value is not valid.
+     */
+    private static void assertAuthor(final String author) {
+        Assert.notNull(author, "The author must not be null");
+    }
+
```

Comment:
Why add this asserts? Is for check null before insert in DB ?

Reply:
Yes. I prefer the models to validate this stuff themselves, and leave the upper layer (a.k.a the service layer) to validate business stuff (for example, age of a user, and that sort of stuff).